pkg/acsengine/engine.go

```diff
@@ -685,8 +685,28 @@ func getDCOSAgentProvisionScript(profile *api.AgentPoolProfile) string {
 		roleFileContents = "touch /etc/mesosphere/roles/slave"
 	}
 
-	provisionScript = strings.Replace(provisionScript, "ROLESFILECONTENTS", roleFileContents, -1)
+	//look for an agent specific script to add
+
+	as, err2 := Asset("dcosagentcustomscript.sh")
+
```

Comment:
I've added PR to ensure a panic does not bring down the app: https://github.com/Azure/acs-engine/pull/218